rules/experimental/rbe/exec_properties.bzl

```diff
@@ -13,6 +13,97 @@
 # limitations under the License.
 
 """This file contains macros to create and manipulate dictionaries of properties to be used as execution properties for RBE.
+
+It also contains macros that create repository rules for standard and custom sets of execution
+property sets.
+
+Here are some examples of how to use these repository rules:
+
+Scenario 1 - The standard use case:
+
+In the WORKSPACE file, call
+  rbe_exec_properties(
+      name = "exec_properties",
+  )
+
+This creates a local repo @rbe_exec_properties with standard RBE execution property sets. For
+example, NETWORK_ON which is the dict {"dockerNetwork" : "standard"}
+
+Then, in some BUILD file, you can reference this execution property set as follows:
+
+  load("@exec_properties//:exec_propert_sets.bzl", "NETWORK_ON")
+  ...
+  exec_properties = NETWORK_ON
+
+The reason not to directly set exec_properties = {...} in a target is that then it might be hard to
+depend on such a target from another repo, if, say, that other repo wants to use remote execution
+but not RBE.
+
+Scenario 2 - local execution
+
+If bazel is set up so that the targets are executed locally, then the contents of exec_properties
+are ignored.
+
+Scenario 3 - non-RBE remote execution:
+
+Let's assume that the non-RBE remote execution endpoint provides a macro similar to
+rbe_exec_properties (say other_re_exec_properties), which populates the same variables (e.g.
+NETWORK_ON) with possibly different dict values.
+In this case, the WORKSPACE would look like this:
+  other_re_exec_properties(
+      name = "exec_properties",
+  )
+
+And the targets in the BUILD files will be able to depend on targets from other repos that were
+written with RBE in mind.
+
+Scenario 4 - rbe_exec_properties with override:
+
+Let's now assume that a particular repo, running with a particular RBE setups, wants to enforce
+for the sake of hermeticity, that no build or test will ever have network access. This would be
```

Comment:
I feel that the word "enforce" is a little too strong. This does not guarantee that there won't be something that turns it on. The rules might e.g., still put this into a dictionary directly.

Maybe "wants to run everything without network access"

Reply:
Done.